src/ff1.rs

```diff
@@ -232,24 +235,35 @@ fn generate_s<'a, CIPH: BlockEncrypt>(
         .take(d)
 }
 
+/// A struct for performing FF1 encryption and decryption operations
+/// using the default 10 Feistel rounds
+pub type FF1<CIPH> = FF1fr<10, CIPH>;
+
+/// A struct for performing hardened FF1 encryption and decryption operations
+/// using 18 Feistel rounds
+pub type FF1h<CIPH> = FF1fr<18, CIPH>;
+
 /// A struct for performing FF1 encryption and decryption operations.
-pub struct FF1<CIPH: BlockCipher> {
+/// with an adjustable number of Feistel rounds
+pub struct FF1fr<const FEISTEL_ROUNDS: u8, CIPH: BlockCipher> {
```

Comment:
Per https://github.com/str4d/fpe/pull/25#issuecomment-1385974102 I do not want to expose the ability to configure an arbitrary numer of rounds, as this includes round numbers that are insecure. If/when Rust gains full const generics, I would consider reopening this, because then we could enforce e.g. `FEISTEL_ROUNDS >= 10`. But for now, this should not be part of the public API:
```suggestion
/// with an adjustable number of Feistel rounds.
struct FF1Core<const FEISTEL_ROUNDS: u8, CIPH: BlockCipher> {
```

Reply:
Why not do this?
```rust
/// FF1 encryption and decryption operations with an adjustable number of Feistel rounds.
/// The number of Feistel rounds will be `10 + EXTRA_ROUNDS_ABOVE_10`.
pub struct FF1WithExtraRounds<const EXTRA_ROUNDS_ABOVE_10: u8, CIPH: BlockCipher> { ... }

/// Hardened FF1 encryption and decryption operations using 18 Feistel rounds.
pub type FF1With18Rounds<CIPH> = FF1WithExtraRounds<8, CIPH>;
```
(poor man's range typing!)

Reply:
That's no way to raise the floor of this bound later if NIST devices to replace FF1 with a higher-round version. So for now I'd prefer to only expose explicit round versions.